File: backend/app/schemas/evidence.py
from datetime import datetime
from typing import Optional, List
from uuid import UUID
from pydantic import BaseModel

from app.models.evidence_file import LinkedType


class EvidenceFileCreate(BaseModel):
    """Schema for creating an evidence file record."""
    linked_type: LinkedType
    linked_id: UUID
    description: Optional[str] = None
    tags: Optional[List[str]] = None


class EvidenceFileResponse(BaseModel):
    """Schema for evidence file response."""
    id: UUID
    filename: str
    storage_path: str
    file_type: Optional[str] = None
    file_size: Optional[str] = None
    linked_type: LinkedType
    linked_id: UUID
    uploaded_by: UUID
    uploaded_at: datetime
    description: Optional[str] = None
    tags: Optional[List[str]] = None

    class Config:
        from_attributes = True


class EvidenceFilter(BaseModel):
    """Schema for filtering evidence files."""
    linked_type: Optional[LinkedType] = None
    linked_id: Optional[UUID] = None

    class Config:
        from_attributes = True

File: frontend/src/types/evidence.ts
export enum LinkedType {
  TRANSACTION = 'transaction',
  PAYROLL = 'payroll',
  PROJECT = 'project',
  ACTIVITY = 'activity',
}

export interface EvidenceFile {
  id: string;
  filename: string;
  storage_path: string;
  file_type: string | null;
  file_size: string | null;
  linked_type: LinkedType;
  linked_id: string;
  uploaded_by: string;
  uploaded_at: string;
  description: string | null;
  tags: string[] | null;
}

export interface EvidenceUploadData {
  file: File;
  linked_type: LinkedType;
  linked_id: string;
  description?: string;
  tags?: string[];
}

export interface EvidenceFilter {
  linked_type?: LinkedType;
  linked_id?: string;
}

export interface PaginatedEvidenceResponse {
  items: EvidenceFile[];
  total: number;
  page: number;
  page_size: number;
}
